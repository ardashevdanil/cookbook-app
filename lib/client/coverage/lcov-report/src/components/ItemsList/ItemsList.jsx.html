<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ItemsList/ItemsList.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/ItemsList</a> ItemsList.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/15</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react';
import './ItemsList.css';
&nbsp;
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCog } from '@fortawesome/free-solid-svg-icons';
&nbsp;
class ItemsList extends React.Component {
  componentDidUpdate(prevProps) {
    const { location, onUpdate } = { ...this.props };
&nbsp;
    // FIX: I think there's a better way to handle location changes
    <span class="missing-if-branch" title="if path not taken" >I</span>if (location.pathname !== prevProps.location.pathname) {
<span class="cstat-no" title="statement not covered" >      if (location.pathname === '/') {</span>
<span class="cstat-no" title="statement not covered" >        onUpdate('recipes');</span>
      } else {
<span class="cstat-no" title="statement not covered" >        onUpdate(location.pathname.slice(1));</span>
      }
    }
  }
&nbsp;
  render() {
    const { items, isFetching, history } = { ...this.props };
    const itemsComponents = items.map(item =&gt; (
      &lt;this.props.item
        content={item}
        history={history}
        key={item.id}
      /&gt;
    ));
&nbsp;
    if (isFetching) {
      return (
        &lt;div className="ItemsList__empty-list"&gt;
          &lt;FontAwesomeIcon
            icon={faCog}
            spin
            size="2x"
          /&gt;
        &lt;/div&gt;
      );
    }
&nbsp;
    if (!items.length) {
      return (
        &lt;div className="ItemsList__empty-list"&gt;
          &lt;div&gt;No Matches&lt;/div&gt;
        &lt;/div&gt;
      );
    }
&nbsp;
    return (
      &lt;div className="ItemsList"&gt;
        { itemsComponents }
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default ItemsList;
&nbsp;
ItemsList.defaultProps = {
  isFetching: true,
  location: {},
};
&nbsp;
ItemsList.propTypes = {
  isFetching: PropTypes.bool,
  items: PropTypes.arrayOf(
    PropTypes.object.isRequired,
  ).isRequired,
  location: PropTypes.object,
  onUpdate: PropTypes.func.isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
