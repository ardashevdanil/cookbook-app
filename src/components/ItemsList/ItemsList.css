.ItemsList {
  margin: 81px 0;
	padding: 18px 33px;
  display: flex;
  flex-wrap: wrap;
  background-color: white;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
}

.ItemsList__empty-list {
  height: 200px;
  margin: 81px 0;
  padding: 18px 33px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: white;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
  font-size: 50px;
}

@media only screen and (max-width: 1120px) {
  .ItemsList {
    padding:  0;
    margin: 0;
    box-shadow: none;
  }

  .ItemsList__empty-list {
    margin: 0;
    padding: 0;
    box-shadow: none;
    font-size: 50px;
  }
}