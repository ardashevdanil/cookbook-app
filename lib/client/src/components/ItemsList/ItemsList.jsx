import PropTypes from 'prop-types';
import React from 'react';
import './ItemsList.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCog } from '@fortawesome/free-solid-svg-icons';

class ItemsList extends React.Component {
  componentDidUpdate(prevProps) {
    const { location, onUpdate } = { ...this.props };

    // FIX: I think there's a better way to handle location changes
    if (location.pathname !== prevProps.location.pathname) {
      if (location.pathname === '/') {
        onUpdate('recipes');
      } else {
        onUpdate(location.pathname.slice(1));
      }
    }
  }

  render() {
    const { items, isFetching, history } = { ...this.props };
    const itemsComponents = items.map(item => (
      <this.props.item
        content={item}
        history={history}
        key={item.id}
      />
    ));

    if (isFetching) {
      return (
        <div className="ItemsList__empty-list">
          <FontAwesomeIcon
            icon={faCog}
            spin
            size="2x"
          />
        </div>
      );
    }

    if (!items.length) {
      return (
        <div className="ItemsList__empty-list">
          <div>No Matches</div>
        </div>
      );
    }

    return (
      <div className="ItemsList">
        { itemsComponents }
      </div>
    );
  }
}

export default ItemsList;

ItemsList.defaultProps = {
  isFetching: true,
  location: {},
};

ItemsList.propTypes = {
  isFetching: PropTypes.bool,
  items: PropTypes.arrayOf(
    PropTypes.object.isRequired,
  ).isRequired,
  location: PropTypes.object,
  onUpdate: PropTypes.func.isRequired,
};
