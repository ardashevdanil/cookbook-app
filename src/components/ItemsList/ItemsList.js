import PropTypes from 'prop-types';
import React from 'react';
import './ItemsList.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCog } from '@fortawesome/free-solid-svg-icons';
import Item from '../Item/Item';
import exampleImg from './exampleImg.png';

class ItemsList extends React.Component {

	/*
	componentDidUpdate(prevProps) {
		if (this.props.location.pathname !== prevProps.location.pathname) {
			
			if (this.props.location.pathname === '/') {
				this.props.onUpdate('home');
			} else {
				this.props.onUpdate(this.props.location.pathname.slice(1));
			}
		}
	}
	*/

	render() {
		const items = fakeData.map( (item, index) => {
			return (
				<Item
					content={ item }
					key={ index }
				/>
			)
		})
		/*
		if (this.props.isFetching) {
			return (
				<div 
					className={ styles.emptyList }
					style={{
						minHeight: document.documentElement.clientHeight - 312
					}}
				>
					<FontAwesomeIcon icon={ faCog } spin size='2x'/>
				</div>
			)
		} else if (!items.length) {
			return (
				<div 
					className={ styles.emptyList }
					style={{
						minHeight: document.documentElement.clientHeight - 312
					}}
				>
					<div>No Matches</div>
				</div>
			)
		}
		*/
	
		return (
			<div 
				className='ItemsList'
			>
				{ items }
			</div>
		);
	}
}

export default ItemsList;

const fakeData = [
	{
		name: 'Raspberry & Cream Frozen Yogurt Pie',
		cookingTime: 35,
		comments: 7,
		watched: 14,
		img: exampleImg,
	},
	{
		name: 'Raspberry & Cream Frozen Yogurt Pie',
		cookingTime: 35,
		comments: 7,
		watched: 14,
		img: exampleImg,
	},
	{
		name: 'Raspberry & Cream Frozen Yogurt Pie',
		cookingTime: 35,
		comments: 7,
		watched: 14,
		img: exampleImg,
	},
	{
		name: 'Raspberry & Cream Frozen Yogurt Pie',
		cookingTime: 35,
		comments: 7,
		watched: 14,
		img: exampleImg,
	},
	{
		name: 'Raspberry & Cream Frozen Yogurt Pie',
		cookingTime: 35,
		comments: 7,
		watched: 14,
		img: exampleImg,
	},
]

/*
ItemsList.propTypes = {
	currency: PropTypes.string.isRequired,
	isFetching: PropTypes.bool,
	items: PropTypes.arrayOf(
		PropTypes.shape({
			img: PropTypes.string,
			name: PropTypes.shape({
				eng: PropTypes.string,
				rus: PropTypes.string,
			}),
			price: PropTypes.shape({
				usd: PropTypes.number,
				rub: PropTypes.number,
			}),
			tags: PropTypes.arrayOf(PropTypes.string)
		})
	).isRequired,
	location: PropTypes.object,
	onUpdate: PropTypes.func.isRequired,
	pushCart: PropTypes.func.isRequired,
	selectedCategory: PropTypes.string.isRequired,
}
*/