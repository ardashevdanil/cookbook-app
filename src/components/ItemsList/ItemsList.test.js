import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import ItemsList from './ItemsList';

Enzyme.configure({ adapter: new Adapter() });

describe('test ItemsList component', () => {
  const testItem = () => null;
	const component = Enzyme.shallow(
		<ItemsList
      isFetching={ false }
      item = { testItem }
      items={[
        {
          comments: 0,
          cookingTime: 0,
          description: '',
          difficult: '',
          id: 0,
          img: '',
          ingridients: ['', ''],
          name: 'test1',
          videoId: '',
          watched: 0,
        },
      ]}
      history={ {} }
      location={{ pathname: '/test' }}
      onUpdate={ () => null }
      selectedCategory='home'
    />
	);

	test('renders correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

  test('should render items', () => {
    expect(component.find(testItem).length).toBe(1);
  });

  test('should render empty list if there aren\'t items', () => {
    component.setProps({ items: [] });

    expect(component.find('div').at(1).text()).toBe('No Matches');
  })
})