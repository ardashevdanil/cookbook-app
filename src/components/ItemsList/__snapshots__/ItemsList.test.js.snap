// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemsList component renders correctly 1`] = `
"<div className=\\"ItemsList\\">
  <testItem content={{...}} history={{...}} />
</div>"
`;
